source/Engine.cpp

```diff
@@ -915,7 +915,8 @@ void Engine::Step(bool isActive)
 		for(const shared_ptr<Minable> &minable : asteroids.Minables())
 		{
 			Point offset = minable->Position() - center;
-			if(offset.Length() > scanRange && flagship->GetTargetAsteroid() != minable)
+			if((offset.Length() > scanRange || !Preferences::Has("Show minable pointers"))
+				&& flagship->GetTargetAsteroid() != minable)
 				continue;
 
 			targets.push_back({
```

Comment:
Instead of checking the preference for every minable within range, put this loop over all the minables inside an if statement checking the preference.

Reply:
Add the targeted asteroid (if there is one) separately.

Reply:
done